### PR TITLE
Experimental

### PR DESCRIPTION
**Pull request by olli991**
Added:
- VIM 
- Universal skin installer
- Brandetection for skins
- Added recovery utility
- Rebuild every script to new logic
- Added mount utility and unmount with mount detector
- Simlyfied some scripts
- Minor bugfixing everywhere

### CODE REVIEW

**Comment by jilleb** [comment]
Great work!
Reviewing + testing today and tomorrow.

**Review by olli991 — COMMENTED** [review]
Some thinks i found which still needs to be fixed, see comments.

File: Toolbox/scripts/util_uninstall.sh [hunk 1]
```diff
@@ -58,13 +63,15 @@ rm -r /mnt/app/eso/hmi/engdefs/scripts/mqb/*.sh
 rm -r /mnt/app/eso/hmi/engdefs/scripts/mqb
 
 # Deleting Toolbox version entry
+export MOUNTPOINT2=4
 echo "Deleting versions entry"
 rm /net/rcc/mnt/efs-persist/SWDL/FileCopyInfo/Toolbox.info
```
[the remark this answers is not in the document]
Comment (olli991) [reply Toolbox/scripts/util_uninstall.sh:68]: Nevermind i defined both mountpoints in the beginning so mount should be active bevore delting the script

**Review by olli991 — COMMENTED** [review]
fixed uninstaller script

File: Toolbox/scripts/util_uninstall.sh [hunk 2]
```diff
@@ -80,13 +87,13 @@ if [ -e ${SSD_INSTALL_DIR} ]; then
...
+	
+	mount -ur /mnt/system
 fi
 
-# Remount as read only
-mount -ur /mnt/app
-mount -ur /mnt/system
-mount -ur /net/rcc/mnt/efs-persist
+# Include back to read-only system mount script
+. /eso/hmi/engdefs/scripts/mqb/util_unmountsys.sh
```
[the remark this answers is not in the document]
Comment (olli991) [reply Toolbox/scripts/util_uninstall.sh]: fixed

File: Toolbox/scripts/flash_ifs.sh [hunk 3]
```diff
@@ -8,20 +9,16 @@ export LD_LIBRARY_PATH=/lib:/mnt/app/root/lib-target:/eso/lib:/mnt/app/usr/lib:/
...
+# Include SD card mount script
 . /eso/hmi/engdefs/scripts/mqb/util_mountsd.sh
-if [[ -z "$VOLUME" ]] 
-then
-	echo "No SD-card found, quitting"
-	exit 0
-fi
-
 
-if test -f "$VOLUME/$SDPATH"; then
```
[the remark this answers is not in the document]
Comment (jilleb) [reply Toolbox/scripts/flash_ifs.sh:24]: Yes, 'test' is nothing. Probably a copy/paste devil from somewhere. But it's good, it prevented people from flashing their IFS :-P

File: Toolbox/scripts/install_lsdjxe.sh [hunk 4]
```diff
@@ -1,47 +1,24 @@
...
-
-#info
+#############################################################################################
+# Info
 export TOPIC=LSD
-export DESCRIPTION="This script will copy custom  lsd.jxe from SD to your unit."
-
-echo $DESCRIPTION
+export MIBPATH=/net/mmx/ifs/lsd.jxe
+export MIBPATH2=/net/mmx/mnt/app
```
[the remark this answers is not in the document]
Comment (jilleb) [reply Toolbox/scripts/install_lsdjxe.sh:8]: path to lsd.jxe is /ifs/
I don't think /net/mmx/ifs is actually existing.. Let's check it on the unit itself.

File: Toolbox/scripts/flash_ifs.sh [hunk 5]
```diff
@@ -8,20 +9,16 @@ export LD_LIBRARY_PATH=/lib:/mnt/app/root/lib-target:/eso/lib:/mnt/app/usr/lib:/
...
 . /eso/hmi/engdefs/scripts/mqb/util_mountsd.sh
-if [[ -z "$VOLUME" ]] 
-then
-	echo "No SD-card found, quitting"
-	exit 0
-fi
-
 
-if test -f "$VOLUME/$SDPATH"; then    
+if [ -f "$VOLUME/$SDPATH" ]; then
```
[the remark this answers is not in the document]
Comment (jilleb) [reply Toolbox/scripts/flash_ifs.sh:21]: Nope :-)

File: Toolbox/scripts/set_displaycontexts.sh [hunk 6]
```diff
@@ -1,28 +0,0 @@
...
-mkdir -p $VOLUME/Dump/$VERSION/$FAZIT/$TOPIC
-/eso/bin/apps/dmdt gc > $DUMPFOLDER/gc.txt
-
-echo "Client information"
-cat /$DUMPFOLDER/gc.txt
-sleep 1
-
-
-
-exit 0
```
[the remark this answers is not in the document]
Comment (jilleb) [reply Toolbox/scripts/set_displaycontexts.sh:28]: Indeed. It was a relic from the past. Something I've used in previous research.

File: Toolbox/scripts/set_displaycontexts_1.sh [hunk 7]
```diff
@@ -1,24 +0,0 @@
...
-#Make dump folder
-
-
-/eso/bin/apps/dmdt sc 0 
-
-sleep 1
-
-
-
-exit 0
```
[the remark this answers is not in the document]
Comment (jilleb) [reply Toolbox/scripts/set_displaycontexts_1.sh:24]: Old stuff :-)

**Pull request merged**